plugins/rikki/heroeslounge/classes/helpers/NotificationHelper.php

```diff
@@ -163,30 +164,43 @@ private static function generateApplicationMessages($a, $sloth, &$retVal)
 
     public static function checkParticipation($sloth, &$retVal)
     {
-        foreach ($sloth->teams as $key => $team) {
-            $s = Seasons::where('is_active', 1)->where('reg_open', 1)->where('region_id', $team->region_id)->whereDoesntHave('teams', function ($q) use ($sloth, $team) {
-                $q->where('team_id', $team->id);
-            })->get();
-            NotificationHelper::generateParticipationMessages($s, $team, $sloth, $retVal);
+        $registerableSeasons = Seasons::where('is_active', 1)->where('reg_open', 1)->where('region_id', $sloth->region_id)->get();
+        foreach ($registerableSeasons as $season) {
+            if ($season->free_agents->contains($sloth->id)) {
+                continue;
+            }
+
+            if ($sloth->isSignedUpForSeason($season)) {
```

Comment:
This function probably wants to be renamed into $isInTeamParticipatingInSeason()